lib/commands/details.rb

```diff
@@ -0,0 +1,64 @@
+# typed: ignore
+# frozen_string_literal: true
+
+require_relative './google_maps_api'
+
+module Commands
+  class Details < GoogleMapsAPI
+    extend T::Sig
+
+    sig { override.returns(String) }
+    def self.name
+      'details'
+    end
+
+    sig { override.returns(String) }
+    def self.help
+      <<~HELP
+        details <place>: returns available info about a specific place listed on Google Maps.
+        If there are more than one match, the command returns a list of the top 10 results instead.
+
+        Examples:
+        details living computers museum, seattle wa
+        details korean restaurant victoria bc
+      HELP
+    end
+
+    sig { override.returns(String) }
+    def response_body
+      return api_key_missing unless api_key_exists?
+
+      return help if arg_text === ''
+
+      begin
+        places = Google::Maps.places(arg_text)
+        place = Google::Maps.place(places.first.place_id)
+      rescue StandardError => error
+        return "NO RESULTS FOR \"#{arg_text}\"" if error.message === 'Google did not return any results: ZERO_RESULTS'
+        return error.message
+      rescue
+        return 'THERE WAS AN ERROR'
+      end
+
+      return "#{places.slice(0, 10).map.with_index { |e, i| "(#{i + 1}) #{e}" }.join(%Q{\n})}" if places.length > 1
```

Comment:
You can optionally pull this out into its own method with a specific name:

```ruby
private

sig { params(places: T::Array[Google::Maps::Place]) }
def list_of_places(places)
  # use `place` as the var name, `e` is not as clear to me
  places.slice(0, 10).map_with_index do |place, i|
    # ...
  end
end
```

and call it
```ruby
return list_of_places(places) if places.length > 1
```

Reply:
I like pulling out chunks of functionality into well-named methods, as you can start reading this method down in more of an "English", conversational kind of way - vs. having to parse what the raw`map_with_index` statement is doing.